### ISSUE TITLE
[Discussion]: Transformation Logging

### ISSUE THREAD

**Issue by c-mueller** [StuPro-TOSCAna/TOSCAna#57]
After investigating the problem of Logging the transformation output for a bit i came across the following solution:

This is based on [Logback](https://logback.qos.ch/) because that's the framework supplied by Spring using it is probably the best solution, it might also be the only one.

However a simililar approach is probably possible with most other Logging Frameworks. One Definately supporting it is the default `java.util.logging` Framework.

The approach goes as Follows:
- The `Transformation` interface has a method called `getTransformationLogger(Class<?> class)`, this method creates a logger that features a special wrting mechanism (called `Appender` in most cases) This method gets called whenever a Specific logger is needed. Mostly while performing a transformation.
	- We could also just return a completely initalized appender that the user will add to the Logger instance. But this seems stupid because some class casting is necessary in order to realize this.
- Except for the different instanciation method (Transformation specific loggers cannot be a static variable) the logger can be used normaly

At least i did not find a way to define something like this using the logging configuration (`logback.xml`) the appenders have to be set programmatically!

**The appender**:
The appender, it will probably be called `TransformationAppender`, will take the logging events and store them somehow. 

I thought about the following approaches:
- The Appender writes the output into a file (specific to the transformation) in the same way that the ConsoleAppender prints it to stdout (System.out). The rest frontend then reads the log file and sends the lines requested by the user, keep in mind we always have to read the complete file event if we only want the last line of a log.
- The appender writes the output into a logfile (just like before) but a copy (list of lines) is kept in memory. The in memory copy will be used to retrieve logs via the rest api.

This currently is incomplete i might add some more ideas later, i just want to hear your thoughts

**Comment by hnicke** [comment]
I'd tackle it this this:
- Each Transformation object has it's Logger Class
- The Logger class stores each log line in memory: List<LogLine> or even List<String>
- The logger class wraps a logback logger, which is programatically setup in order to write to a custom logfile (like chris mentioned)
- when logger.info() etc is called, the logline is written to memory, then super.info() etc is called --> logline gets written to file, too
- the rest controller can then operate on the in-memory logger object, while persistency is also assured

**Comment by c-mueller** [comment]
In the discussion with @hnicke yesterday we decided to implement this in the way described above, using Logback. The Logger will store the logs in memory, for quick querying from the api and it will store a copy of the logs in a special log file. The in memory appender has been implemented in the commit e8cd102362033865080a159e62eb18b2a271d8be. The file appender will be added once the CsarRepository is able to to provide a filepath to write the logs to. A seperate issue will be created for that.

This issue will be closed after the next commit
